### PR TITLE
#167313400 Users should be able to reset password via email

### PR DESCRIPTION
**Pull request by minega25**
#### What does this PR do?
- Users should be able to reset password via email
#### Description of Task to be completed?
- Password reset functionality is missing in our application. Its invaluable in instances where our users forget their login credentials
#### How should this be manually tested?
- `git checkout -b ft-user-password-reset-167313400`
- `git pull origin ft-user-password-reset-167313400`
- `npm install`
- `npm run dev`
- Using POST method, use `api/v1/users/reset` endpoint with email attribute in req.body the user must be registered
- A password reset email having a reset password link is sent to the owner of the email address used above
- Copy the link in the email body then paste in it postman.
- Using the PATCH method make sure you send the request having in the req.body `password and confirmPassword`.

#### What are the relevant pivotal tracker stories?
[#167313400](https://www.pivotaltracker.com/story/show/167313400)
#### Screenshots (if appropriate)
Screen Shot 2019-08-13 at 18 58 34

### CODE REVIEW

**Review by salviosage — APPROVED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -12,7 +12,6 @@
     "coveralls": "nyc report --reporter=text-lcov | coveralls",
     "test": "export NODE_ENV=test && npm run undomigration && npm run migration && npm run seeds && nyc --reporter=html --reporter=text mocha ./test --no-timeout --exit --require @babel/register",
     "dev": "npm run undoseeds && npm run seeds && nodemon --exec babel-node ./src/app.js"
```
Comment (nshutijonathan) [inline package.json]: @patric is this "npm  dev" working as expected?

**Review by nshutijonathan — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (minega25) [reply package.json]: No. I did not want to change it before informing the team

Comment (Mnickii) [reply package.json]: @minega25 it's missing the ```npm run migration``` after undoseeds.

**Review by Mnickii — CHANGES_REQUESTED** [review]

**Review by nkalyesubula — CHANGES_REQUESTED** [review]
Nice work Patrick. Though I think a user copying the link sent to him on email and pasting it in postman just to change is password is cumbersome work, more so if the user is not that technical. I would suggest you refactor this, and maybe provide a temporary password to the user when the link is clicked so that he can change it later when after signing in. Also fix the npm run dev

**Comment by minega25** [comment]
@Mnickii What would be the best approach for this feature? According to @nkalyesubula's feedback?

**Review by habinezadalvan — APPROVED** [review]
Good work.

**Comment by Mnickii** [comment]
@nkalyesubula @minega25 the feature states that a user should be able to reset the password via email. As a user, I'd expect a link sent to my email, that I can follow to reset my password. In this way, I find that the implementation has been achieved. providing a temporary password has it's own security issues.

File: package.json [hunk 2]
```diff
@@ -11,8 +11,7 @@
     "undomigration": "export NODE_ENV=test && npm run undoseeds && babel-node node_modules/.bin/sequelize db:migrate:undo:all",
     "coveralls": "nyc report --reporter=text-lcov | coveralls",
     "test": "export NODE_ENV=test && npm run undomigration && npm run migration && npm run seeds && nyc --reporter=html --reporter=text mocha ./test --no-timeout --exit --require @babel/register",
-    "dev": "npm run undoseeds && npm run seeds && nodemon --exec babel-node ./src/app.js"
-  
+    "dev": "nodemon --exec babel-node ./src/app.js"
   },
```
Comment (Mnickii) [inline package.json:16]: maybe then here you can add a script to include all steps in migrations like``` "runmigrations": "npm run undoseeds && npm run migration && npm run seeds"```

**Review by Mnickii — COMMENTED** [review]
@minega25

**Review by nkalyesubula — APPROVED** [review]

**Pull request merged**